base/random.jl

```diff
@@ -1224,6 +1227,14 @@ let Floats = Union{Float16,Float32,Float64}
             $randfun(                                dims::Integer...               ) = $randfun(GLOBAL_RNG, Float64, dims...)
         end
     end
+
+    for (T,SQRT_HALF) in ((Float64, 0.7071067811865475),
+                          (Float32, 0.70710677f0),
+                          (Float16, 0.70710677f0))
+        @inline function randn(rng::AbstractRNG, ::Type{Complex{T}})
+            Complex{T}(SQRT_HALF*randn(rng, T), SQRT_HALF*randn(rng, T))
+        end
+    end
```

Comment:
You could also write `sqrt(T(2))`: the compiler should be smart enough to inline this.